__tests__/app.test.js

```diff
@@ -116,6 +132,41 @@ describe("GET /api/articles/:article_id", () => {
   });
 });
 
+describe("GET /api/articles/:article_id/comments", () => {
+  test("200: Returns all comments based on article_id, descending order", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body: { comments } }) => {
+          expect(comments.length).toBeGreaterThan(0);
+
+          expect(comments[0].article_id).toBe(1);
+
+          expect(comments).toBeSorted({ descending: true });
+
+          comments.forEach((comment) => {
+            expect(comment).toEqual(
+              expect.objectContaining({
+                comment_id: expect.any(Number),
+                votes: expect.any(Number),
+                created_at: expect.any(String),
+                author: expect.any(String),
+                body: expect.any(String),
+                article_id: expect.any(Number),
+            }))
+          })
+        })
+      });
+  test("204: returns no comments found when using a valid article_id that has no comments", () => {
+    return request(app)
+      .get("/api/articles/23425/comments")
```

Comment:
also this article doesn't actually exist - so we aren't getting comments because there are no comments for this article, but because the article actually doesn't exist - so this is a separate point we also have to test for - also if the article id is invalid, we should check for that too!